### PR TITLE
Align qpl_deflate codec with master

### PR DESCRIPTION
**Pull request by jinjunzh**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Aligned qpl_deflate codec with changes in master [57291](https://github.com/ClickHouse/ClickHouse/pull/57291)
- Bumped Intel QPL (used by codec `DEFLATE_QPL`) from v1.2.0 to v1.3.1 .
- Fixed a bug in case of BOF (Block On Fault) = 0, changed to handle page faults by falling back to SW path.

### CODE REVIEW

File: contrib/idxd-config-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -8,16 +8,16 @@ set (SRCS
     "${LIBACCEL_SOURCE_DIR}/util/sysfs.c"
 )
 
-add_library(accel-config ${SRCS})
+add_library(_accel-config ${SRCS})
```
Comment (alexey-milovidov) [inline contrib/idxd-config-cmake/CMakeLists.txt:11]: Why?

File: contrib/qpl-cmake/CMakeLists.txt [hunk 2]
```diff
@@ -300,48 +303,48 @@ set_target_properties(middle_layer_lib PROPERTIES CXX_STANDARD 17)
...
 # [SUBDIR]c_api
-file(GLOB_RECURSE QPL_C_API_SRC 
+file(GLOB_RECURSE QPL_C_API_SRC
         ${QPL_SRC_DIR}/c_api/*.c
         ${QPL_SRC_DIR}/c_api/*.cpp)
 
 get_property(LIB_DEPS GLOBAL PROPERTY QPL_LIB_DEPS)
 
-add_library(qpl STATIC ${QPL_C_API_SRC} ${LIB_DEPS})
+add_library(_qpl STATIC ${QPL_C_API_SRC} ${LIB_DEPS})
```
Comment (alexey-milovidov) [inline contrib/qpl-cmake/CMakeLists.txt:320]: Why?

File: src/Compression/CompressionCodecDeflateQpl.h [hunk 3]
```diff
@@ -108,8 +114,8 @@ class CompressionCodecDeflateQpl : public ICompressionCodec
 private:
     UInt32 getMaxCompressedDataSize(UInt32 uncompressed_size) const override;
 
-    std::unique_ptr<HardwareCodecDeflateQpl> hw_codec;
     std::unique_ptr<SoftwareCodecDeflateQpl> sw_codec;
+    std::unique_ptr<HardwareCodecDeflateQpl> hw_codec;
```
Comment (alexey-milovidov) [inline src/Compression/CompressionCodecDeflateQpl.h:118]: Why?

**Review by alexey-milovidov — APPROVED** [review]

File: contrib/idxd-config-cmake/CMakeLists.txt (hunk 1 above)
Comment (jinjunzh) [reply contrib/idxd-config-cmake/CMakeLists.txt:11]: Just want to align with [CMakeLists.txt in master](https://github.com/ClickHouse/ClickHouse/blob/master/contrib/idxd-config-cmake/CMakeLists.txt)

File: contrib/qpl-cmake/CMakeLists.txt (hunk 2 above)
Comment (jinjunzh) [reply contrib/qpl-cmake/CMakeLists.txt:320]: Just want to align with [CMakeLists.txt in master](https://github.com/ClickHouse/ClickHouse/blob/master/contrib/qpl-cmake/CMakeLists.txt)

File: src/Compression/CompressionCodecDeflateQpl.h (hunk 3 above)
Comment (jinjunzh) [reply src/Compression/CompressionCodecDeflateQpl.h:118]: The point of switching the order of members in the class was that we could continue to initialize them in the initialization list
[PR in master](https://github.com/ClickHouse/ClickHouse/pull/57291#discussion_r1408928477)

`CompressionCodecDeflateQpl::CompressionCodecDeflateQpl()
    : sw_codec(std::make_unique<SoftwareCodecDeflateQpl>())
    , hw_codec(std::make_unique<HardwareCodecDeflateQpl>(sw_codec))
{
    ...`

**Comment by alexey-milovidov** [comment]
I didn't notice it was a PR to a branch. Then it does not matter.

**Pull request merged**